[CI] DynamicMappingIT testDenseVectorDynamicMapping failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #8946 / almalinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/cjb7v5v6lwnlm)
- [elasticsearch-pull-request #77136 / part-1](https://gradle-enterprise.elastic.co/s/i2stfuupdflp2)

**Reproduction Line:**
```
./gradlew ":server:internalClusterTest" --tests "org.elasticsearch.index.mapper.DynamicMappingIT.testDenseVectorDynamicMapping" -Dtests.seed=EA60295F92FBDEDF -Dtests.locale=th-TH-u-nu-thai-x-lvariant-TH -Dtests.timezone=Europe/Jersey -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.index.mapper.DynamicMappingIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testDenseVectorDynamicMapping),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: null
```

**Issue Reasons:**
- [main] 2 failures in test testDenseVectorDynamicMapping (12.5% fail rate in 16 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -568,9 +568,6 @@ tests:
 - class: org.elasticsearch.index.mapper.vectors.DenseVectorFieldMapperTests
   method: testExistsQueryMinimalMapping
   issue: https://github.com/elastic/elasticsearch/issues/129911
-- class: org.elasticsearch.index.mapper.DynamicMappingIT
-  method: testDenseVectorDynamicMapping
-  issue: https://github.com/elastic/elasticsearch/issues/129928
 
 # Examples:
 #

--- a/server/src/internalClusterTest/java/org/elasticsearch/index/mapper/DynamicMappingIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/index/mapper/DynamicMappingIT.java
@@ -924,15 +924,21 @@ public class DynamicMappingIT extends ESIntegTestCase {
         client().index(
             new IndexRequest("test").source("vector_bbq", Randomness.get().doubles(BBQ_DIMS_DEFAULT_THRESHOLD, 0.0, 5.0).toArray())
         ).get();
-        Map<String, Object> mappings = indicesAdmin().prepareGetMappings(TEST_REQUEST_TIMEOUT, "test")
-            .get()
-            .mappings()
-            .get("test")
-            .sourceAsMap();
-        assertTrue(new WriteField("properties.vector_int8", () -> mappings).exists());
-        assertTrue(new WriteField("properties.vector_int8.index_options.type", () -> mappings).get(null).toString().equals("int8_hnsw"));
-        assertTrue(new WriteField("properties.vector_bbq", () -> mappings).exists());
-        assertTrue(new WriteField("properties.vector_bbq.index_options.type", () -> mappings).get(null).toString().equals("bbq_hnsw"));
+
+        assertBusy(() -> {
+            Map<String, Object> mappings = indicesAdmin().prepareGetMappings(TEST_REQUEST_TIMEOUT, "test")
+                .get()
+                .mappings()
+                .get("test")
+                .sourceAsMap();
+
+            assertTrue(new WriteField("properties.vector_int8", () -> mappings).exists());
+            assertTrue(
+                new WriteField("properties.vector_int8.index_options.type", () -> mappings).get(null).toString().equals("int8_hnsw")
+            );
+            assertTrue(new WriteField("properties.vector_bbq", () -> mappings).exists());
+            assertTrue(new WriteField("properties.vector_bbq.index_options.type", () -> mappings).get(null).toString().equals("bbq_hnsw"));
+        });
     }
 
     public void testBBQDynamicMappingWhenFirstIngestingDoc() throws Exception {
@@ -956,12 +962,15 @@ public class DynamicMappingIT extends ESIntegTestCase {
 
         client().index(new IndexRequest("test").source("vector", Randomness.get().doubles(BBQ_DIMS_DEFAULT_THRESHOLD, 0.0, 5.0).toArray()))
             .get();
-        Map<String, Object> updatedMappings = indicesAdmin().prepareGetMappings(TEST_REQUEST_TIMEOUT, "test")
-            .get()
-            .mappings()
-            .get("test")
-            .sourceAsMap();
-        assertTrue(new WriteField("properties.vector", () -> updatedMappings).exists());
-        assertTrue(new WriteField("properties.vector.index_options.type", () -> updatedMappings).get(null).toString().equals("bbq_hnsw"));
+
+        assertBusy(() -> {
+            Map<String, Object> updatedMappings = indicesAdmin().prepareGetMappings(TEST_REQUEST_TIMEOUT, "test")
+                .get()
+                .mappings()
+                .get("test")
+                .sourceAsMap();
+            assertTrue(new WriteField("properties.vector", () -> updatedMappings).exists());
+            assertTrue(new WriteField("properties.vector.index_options.type", () -> updatedMappings).get("").toString().equals("bbq_hnsw"));
+        });
     }
 }

--- a/server/src/internalClusterTest/java/org/elasticsearch/index/mapper/DynamicMappingIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/index/mapper/DynamicMappingIT.java
@@ -920,9 +920,11 @@ public class DynamicMappingIT extends ESIntegTestCase {
 
         client().index(
             new IndexRequest("test").source("vector_int8", Randomness.get().doubles(BBQ_DIMS_DEFAULT_THRESHOLD - 1, 0.0, 5.0).toArray())
+                .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
         ).get();
         client().index(
             new IndexRequest("test").source("vector_bbq", Randomness.get().doubles(BBQ_DIMS_DEFAULT_THRESHOLD, 0.0, 5.0).toArray())
+                .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
         ).get();
 
         assertBusy(() -> {
@@ -960,8 +962,10 @@ public class DynamicMappingIT extends ESIntegTestCase {
         assertTrue(new WriteField("properties.vector", () -> mappings).exists());
         assertFalse(new WriteField("properties.vector.index_options.type", () -> mappings).exists());
 
-        client().index(new IndexRequest("test").source("vector", Randomness.get().doubles(BBQ_DIMS_DEFAULT_THRESHOLD, 0.0, 5.0).toArray()))
-            .get();
+        client().index(
+            new IndexRequest("test").source("vector", Randomness.get().doubles(BBQ_DIMS_DEFAULT_THRESHOLD, 0.0, 5.0).toArray())
+                .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
+        ).get();
 
         assertBusy(() -> {
             Map<String, Object> updatedMappings = indicesAdmin().prepareGetMappings(TEST_REQUEST_TIMEOUT, "test")

